test/register.test.js

```diff
@@ -98,5 +84,45 @@ function makeRequest (path) {
       t.strictEqual(response.headers['content-length'], '' + body.length)
       t.deepEqual(JSON.parse(body), { hello: 'world' })
     })
+  }
+})
+
+test('internal route declaration should pass the error generated by the register to the next handler / 1', t => {
+  t.plan(1)
+  const fastify = Fastify()
+
+  fastify.register((instance, opts, next) => {
+    next(new Error('kaboom'))
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.send({ hello: 'world' })
+  })
+
+  fastify.listen(0, err => {
+    fastify.close()
+    t.is(err.message, 'kaboom')
   })
-}
+})
+
+test('internal route declaration should pass the error generated by the register to the next handler / 2', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  fastify.register((instance, opts, next) => {
+    next(new Error('kaboom'))
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.send({ hello: 'world' })
+  })
+
+  fastify.after(err => {
+    t.is(err.message, 'kaboom')
```

Comment:
Could you use the error equality?

```t.is(err, callbackError)```

where callbackError is the created error in the plugin

Reply:
I could be wrong, but `is` is an alias of `equal`, and you can't compare objects (like errors) with `equal`, but you should use `deepEqual`.

By the way we use `err.message` in many places along our test, I think is fine.